webextension/js/popup.js

```diff
@@ -3,31 +3,37 @@ const CONTAINER_HIDE_SRC = '/img/container-hide.svg';
 const CONTAINER_UNHIDE_SRC = '/img/container-unhide.svg';
 
 function showOrHideContainerTabs(userContextId, hasHiddenTabs) {
-  return new Promise((resolve, reject) => {
-    const hideorshowIcon = document.querySelector(`#uci-${userContextId}-hideorshow-icon`);
-
-    browser.runtime.sendMessage({
-      method: hasHiddenTabs ? 'showTabs' : 'hideTabs',
+  // Let's show/hide the tabs
+  return browser.runtime.sendMessage({
+    method: hasHiddenTabs ? 'showTabs' : 'hideTabs',
+    userContextId: userContextId
+  })
+  // We need to retrieve the new identity configuration in order to choose the
+  // correct icon.
+  .then(() => {
+    return browser.runtime.sendMessage({
+      method: 'getIdentity',
       userContextId: userContextId
-    }).then(() => {
-      return browser.runtime.sendMessage({
-        method: 'getIdentity',
-        userContextId: userContextId
-      });
-    }).then((identity) => {
-      if (!identity.hasHiddenTabs && !identity.hasOpenTabs) {
-        hideorshowIcon.style.display = "none";
-      } else {
-        hideorshowIcon.style.display = "";
-      }
+    });
+  })
+  // Let's update the icon.
+  .then((identity) => {
+    let hideorshowIcon = document.querySelector(`#uci-${identity.userContextId}-hideorshow-icon`);
+    if (!identity.hasHiddenTabs && !identity.hasOpenTabs) {
+      hideorshowIcon.style.display = "none";
+    } else {
+      hideorshowIcon.style.display = "";
+    }
 
-      hideorshowIcon.src = hasHiddenTabs ? CONTAINER_HIDE_SRC : CONTAINER_UNHIDE_SRC;
-    }).then(resolve);
+    hideorshowIcon.src = hasHiddenTabs ? CONTAINER_HIDE_SRC : CONTAINER_UNHIDE_SRC;
+
+    // The new identity is returned.
+    return identity;
   });
 }
 
 if (localStorage.getItem('onboarded2')) {
-  for (const element of document.querySelectorAll('.onboarding')) {
+  for (let element of document.querySelectorAll('.onboarding')) {
```

Comment:
const is for immutable objects. We are not actually changing element, but to me seems wrong that we change attributes in element when it is const. Maybe because I write too much C++.